### PR TITLE
feat(azure): create arm check StorageAccountMinimumTlsVersion CKV_AZURE_236

### PR DESCRIPTION
**Pull request by OfekShimko**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    #    
    Allowed prefixs:
    ansible|argo|arm|azure|bicep|bitbucket|circleci|cloudformation|dockerfile|github|gha|gitlab|helm|kubernetes|kustomize|openapi|sast|sca|secrets|serverless|terraform|general|graph|terraform_plan|terraform_json
    #
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description
Creating CKV_AZURE_236,  arm networking check.
Looks for minimumTlsVersion configuration to be set to TLS1_2

Fixes # (issue)

### Description
*Include a description of what makes it a violation and any relevant external links.*

### Fix
*How does someone fix the issue in code and/or in runtime?*

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my feature, policy, or fix is effective and works
- [ ] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by JamesWoolfenden** [comment]
You might want to have a test case where minimumTlsVersion hasn't been set.

**Comment by OfekShimko** [comment]
> You might want to have a test case where minimumTlsVersion hasn't been set.

Thank you! Added

**Review by Saarett — APPROVED** [review]
💯

**Review by Eliran-Turgeman — APPROVED** [review]

File: checkov/arm/checks/resource/StorageAccountMinimumTlsVersion.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        id = "CKV_AZURE_236"
+        supported_resources = ('Microsoft.Storage/storageAccounts',)
+        categories = (CheckCategories.NETWORKING,)
+
+        super().__init__(name=name, id=id, categories=categories, supported_resources=supported_resources)
+
+    def scan_resource_conf(self, conf: dict[str, Any]) -> CheckResult:
+        if "properties" in conf and \
+            "minimumTlsVersion" in conf["properties"] and \
+                conf["properties"]["minimumTlsVersion"] in ['TLS1_2']:
```
[the remark this answers is not in the document]
Comment (OfekShimko) [reply checkov/arm/checks/resource/StorageAccountMinimumTlsVersion.py:27]: Thank you!

**Pull request merged**